### ISSUE TITLE
Compatibility with Visium HD data

### ISSUE THREAD

**Issue by cherrie-g** [satijalab/seurat#8700]
Hi there,

I‘m wondering if the Seurat package compatible with the new Visium HD data. Actually, I saw the demo data on 10x genomics website and downloaded to local. But when I tried to load the HD spatial data as normal, I found the image were not import correctly. Like this:
![image](https://github.com/satijalab/seurat/assets/66763863/c627b40d-5661-49d4-a4e6-976b0bba608e)

And I went back the `spatial` directory and found that the `tissue_positions_list.csv` file was replaced by a `.parquet` file. 
![image](https://github.com/satijalab/seurat/assets/66763863/2150b7a6-cdae-4886-a9ed-38a47d72bb2a)

I wonder if there is a solution in currrent version?  My `sessionInfo` is like below. But I don't think it's the version problem.
```
R version 4.1.1 (2021-08-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22621)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.936 
[2] LC_CTYPE=Chinese (Simplified)_China.936   
[3] LC_MONETARY=Chinese (Simplified)_China.936
[4] LC_NUMERIC=C                              
[5] LC_TIME=Chinese (Simplified)_China.936    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.4.2      SeuratObject_5.0.1 Seurat_4.3.0      

loaded via a namespace (and not attached):
  [1] Rtsne_0.16             colorspace_2.1-0       deldir_1.0-6          
  [4] ellipsis_0.3.2         ggridges_0.5.4         IRdisplay_1.1         
  [7] base64enc_0.1-3        spatstat.data_3.0-1    farver_2.1.1          
 [10] leiden_0.4.3           listenv_0.9.0          bit64_4.0.5           
 [13] ggrepel_0.9.3          fansi_1.0.4            codetools_0.2-18      
 [16] splines_4.1.1          polyclip_1.10-4        IRkernel_1.3.2        
 [19] spam_2.10-0            jsonlite_1.8.4         ica_1.0-3             
 [22] cluster_2.1.2          png_0.1-8              uwot_0.1.14           
 [25] shiny_1.7.4            sctransform_0.3.5      spatstat.sparse_3.0-1 
 [28] compiler_4.1.1         httr_1.4.5             Matrix_1.6-5          
 [31] fastmap_1.1.1          lazyeval_0.2.2         cli_3.6.1             
 [34] later_1.3.0            htmltools_0.5.5        tools_4.1.1           
 [37] igraph_1.4.2           dotCall64_1.1-1        gtable_0.3.3          
 [40] glue_1.6.2             RANN_2.6.1             reshape2_1.4.4        
 [43] dplyr_1.1.2            Rcpp_1.0.10            scattermore_0.8       
 [46] vctrs_0.6.2            spatstat.explore_3.1-0 nlme_3.1-152          
 [49] progressr_0.14.0       lmtest_0.9-40          spatstat.random_3.1-4 
 [52] stringr_1.5.0          globals_0.16.2         mime_0.12             
 [55] miniUI_0.1.1.1         lifecycle_1.0.4        irlba_2.3.5.1         
 [58] goftest_1.2-3          future_1.33.1          MASS_7.3-54           
 [61] zoo_1.8-12             scales_1.2.1           promises_1.2.0.1      
 [64] spatstat.utils_3.0-2   parallel_4.1.1         RColorBrewer_1.1-3    
 [67] reticulate_1.28        pbapply_1.7-0          gridExtra_2.3         
 [70] stringi_1.7.12         repr_1.1.6             rlang_1.1.0           
 [73] pkgconfig_2.0.3        matrixStats_0.63.0     evaluate_0.23         
 [76] lattice_0.20-44        ROCR_1.0-11            purrr_1.0.1           
 [79] tensor_1.5             patchwork_1.1.2        htmlwidgets_1.6.2     
 [82] bit_4.0.5              cowplot_1.1.1          tidyselect_1.2.0      
 [85] parallelly_1.36.0      RcppAnnoy_0.0.20       plyr_1.8.8            
 [88] magrittr_2.0.3         R6_2.5.1               generics_0.1.3        
 [91] pbdZMQ_0.3-10          withr_2.5.0            pillar_1.9.0          
 [94] fitdistrplus_1.1-8     survival_3.2-11        abind_1.4-5           
 [97] sp_2.1-2               tibble_3.2.1           future.apply_1.11.1   
[100] hdf5r_1.3.8            crayon_1.5.2           uuid_1.1-1            
[103] KernSmooth_2.23-20     utf8_1.2.3             spatstat.geom_3.1-0   
[106] plotly_4.10.1          grid_4.1.1             data.table_1.14.8     
[109] digest_0.6.31          xtable_1.8-4           tidyr_1.3.0           
[112] httpuv_1.6.9           munsell_0.5.0          viridisLite_0.4.1
```

**Comment by lilwo** [comment]
Hi ,
I converted the tissue_positions.parquet format to csv  with python : pyarrow.parquet, and then re-edited the Read10X_Image function.But I had a new problem, the barcode didn't match in img object . If i use match ,there is fewer barcode , and I didn't know why,
![1711705766132(1)](https://github.com/satijalab/seurat/assets/88726188/c7354d85-f649-448a-849d-01b720bfb866)
![1711705829785](https://github.com/satijalab/seurat/assets/88726188/d4b3f695-c68d-4d2b-b7f4-d85cf92cef30)
`> sessionInfo()
R version 4.0.5 (2021-03-31)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.4.3       png_0.1-7           hdf5r_1.3.5        
[4] cowplot_1.1.1       pheatmap_1.0.12     SeuratObject_4.0.4 
[7] Seurat_4.1.0        VisiumSpatial_0.1.0

loaded via a namespace (and not attached):
  [1] shadowtext_0.0.8       fastmatch_1.1-0        VGAM_1.1-6            
  [4] plyr_1.8.6             igraph_1.2.6           lazyeval_0.2.2        
  [7] GSEABase_1.52.1        splines_4.0.5          densityClust_0.3.2    
 [10] BiocParallel_1.24.1    listenv_0.8.0          scattermore_0.7       
 [13] fastICA_1.2-3          digest_0.6.27          htmltools_0.5.7       
 [16] GOSemSim_2.16.1        viridis_0.6.2          GO.db_3.12.1          
 [19] fansi_0.4.2            magrittr_2.0.1         memoise_2.0.0         
 [22] tensor_1.5             cluster_2.1.2          ROCR_1.0-11           
 [25] limma_3.46.0           globals_0.16.2         annotate_1.68.0       
 [28] graphlayouts_0.7.1     matrixStats_0.58.0     docopt_0.7.1          
 [31] spatstat.sparse_3.0-0  enrichplot_1.10.2      colorspace_2.0-1      
 [34] blob_1.2.1             ggrepel_0.9.1          dplyr_1.0.6           
 [37] crayon_1.4.1           sparsesvd_0.2          jsonlite_1.7.2        
 [40] graph_1.68.0           scatterpie_0.1.6       spatstat.data_3.0-0   
 [43] survival_3.2-11        zoo_1.8-10             glue_1.4.2            
 [46] polyclip_1.10-0        gtable_0.3.0           leiden_0.3.7          
 [49] future.apply_1.7.0     BiocGenerics_0.36.1    abind_1.4-5           
 [52] scales_1.2.1           DOSE_3.16.0            DBI_1.1.1             
 [55] miniUI_0.1.1.1         Rcpp_1.0.11            viridisLite_0.4.0     
 [58] xtable_1.8-4           reticulate_1.20        spatstat.core_2.1-2   
 [61] bit_4.0.4              stats4_4.0.5           htmlwidgets_1.6.4     
 [64] httr_1.4.2             fgsea_1.16.0           FNN_1.1.3             
 [67] RColorBrewer_1.1-2     ellipsis_0.3.2         ica_1.0-2             
 [70] pkgconfig_2.0.3        XML_3.99-0.6           farver_2.1.0          
 [73] uwot_0.1.10            deldir_1.0-6           utf8_1.2.1            
 [76] labeling_0.4.2         tidyselect_1.2.0       rlang_1.1.1           
 [79] reshape2_1.4.4         later_1.2.0            AnnotationDbi_1.52.0  
 [82] munsell_0.5.0          tools_4.0.5            cachem_1.0.5          
 [85] downloader_0.4         cli_3.6.1              generics_0.1.0        
 [88] RSQLite_2.2.7          ggridges_0.5.3         stringr_1.4.0         
 [91] fastmap_1.1.0          goftest_1.2-2          bit64_4.0.5           
 [94] fitdistrplus_1.1-3     tidygraph_1.2.0        DDRTree_0.1.5         
 [97] purrr_1.0.2            RANN_2.6.1             ggraph_2.0.5          
[100] pbapply_1.4-3          future_1.30.0          nlme_3.1-152          
[103] mime_0.10              slam_0.1-50            monocle_2.18.0        
[106] ggrastr_1.0.2          DO.db_2.9              rstudioapi_0.13       
[109] compiler_4.0.5         beeswarm_0.4.0         plotly_4.9.3          
[112] spatstat.utils_3.0-1   tibble_3.2.1           tweenr_1.0.2          
[115] stringi_1.6.2          lattice_0.20-44        Matrix_1.5-4          
[118] HSMMSingleCell_1.10.0  vctrs_0.6.5            pillar_1.9.0          
[121] lifecycle_1.0.3        BiocManager_1.30.15    combinat_0.0-8        
[124] spatstat.geom_3.0-6    lmtest_0.9-38          RcppAnnoy_0.0.18      
[127] data.table_1.14.0      irlba_2.3.3            httpuv_1.6.1          
[130] patchwork_1.1.1        qvalue_2.22.0          R6_2.5.0              
[133] promises_1.2.0.1       KernSmooth_2.23-20     gridExtra_2.3         
[136] vipor_0.4.5            IRanges_2.24.1         parallelly_1.33.0     
[139] codetools_0.2-18       MASS_7.3-54            assertthat_0.2.1      
[142] withr_2.5.0            qlcMatrix_0.9.7        sctransform_0.3.3     
[145] S4Vectors_0.28.1       mgcv_1.8-35            parallel_4.0.5        
[148] clusterProfiler_3.18.1 grid_4.0.5             rpart_4.1-15          
[151] tidyr_1.1.3            rvcheck_0.1.8          Rtsne_0.16            
[154] ggforce_0.3.3          Biobase_2.50.0         shiny_1.6.0           
[157] ggbeeswarm_0.6.0      
`

**Comment by dcollins15** [comment]
Please follow the install instructions included our [quickstart vignette](https://satijalab.org/seurat/articles/visiumhd_commands_intro) - support for Visium HD will be introduced in v5.1.0 but for now is available as pre-release v5.0.3.9909 👍
